src/reducers/CompareResults.ts

```diff
@@ -24,6 +26,31 @@ const compareResults = createSlice({
       state.data = comparisonResults;
     },
   },
+  extraReducers: (builder) => {
+    builder
+      .addCase(fetchCompareResults.fulfilled, (state, action) => {
```

Comment:
I'm encountering issues here. When we select 3 new revisions, there's 3 calls to the backend and I'd like to store the results in state - but when I want to use this state I get error "Uncaught TypeError: state.compareResults.results[0] is undefined".

Reply:
Hi Beatrice, you need to use a `useEffect `because the data hasn't loaded yet, that's why it's undefined. 

So in `src/components/CompareResults/beta/ResultsTable.tsx`
try instead:

 ```
 const results = useAppSelector(
    (state: RootState) => state.compareResults.results,
  );

  const [loading, setLoading] = useState(true);
  const [compareData, setCompareData] = useState<CompareResultsItem[]>([]);

  useEffect(() => {
    if (results.length > 0) {
      setCompareData(results[0].data);
      setLoading(false);
    }
  }, [results]);

  const processedResults = processResults(compareData);
```

and below:

```
         {loading ? (
            <caption>Loading...</caption>
          ) : (
            processedResults.map((res, index) => (
              <TableContent
                themeMode={themeMode}
                key={index}
                header={res.revisionHeader}
                results={res.value}
              />
            ))
          )}
```

Reply:
We will need a loader to give the user feedback while they wait instead of the "No results" component.